stage1/init/common/pod.go

```diff
@@ -692,11 +692,19 @@ func GetFlavor(p *stage1commontypes.Pod) (flavor string, systemdVersion int, err
 	}
 	systemdVersionString := strings.Trim(string(systemdVersionBytes), " \n")
 
-	// systemdVersionString is of the form v229, remove the first character to
-	// get the number
+	// systemdVersionString is either a tag name or a branch name. If it's a
+	// tag name it's of the form "v229", remove the first character to get the
+	// number.
 	systemdVersion, err = strconv.Atoi(systemdVersionString[1:])
 	if err != nil {
-		return "", -1, errwrap.Wrap(errors.New("error parsing stage1's systemd version"), err)
+		// If we get a syntax error, it means the parsing of the version string
+		// of the form "v229" failed, assume it is a branch name and the
+		// systemd version is at least v227
+		if e, ok := err.(*strconv.NumError); ok && e.Err == strconv.ErrSyntax {
+			systemdVersion = 227
```

Comment:
I don't really like hardcoding 227 here...

Instead, could GetFlavor() just return systemdVersion=0, and document that it means systemd version cannot be guessed? Then the callers of GetFlavor() would handle that.

Alternatively, stage1/usr_from_src/usr_from_src.mk could just detect that the version is not of the form v229 (see UFS_SYSTEMD_TAG_MATCH) and read systemd's configure.ac instead... But the first solution seems easier.